code/game/objects/items/weapons/tools/weldingtool.dm

```diff
@@ -98,14 +98,14 @@
 		src.add_fingerprint(user)
 		return
 
-	if((!status) && (istype(W,/obj/item/stack/material/rods)))
+	if(!status && istype(W, /obj/item/stack/material/rods))
 		var/obj/item/stack/material/rods/R = W
 		R.use(1)
-		var/obj/item/flamethrower/F = new/obj/item/flamethrower(user.loc)
-		user.drop_from_inventory(src, F)
-		F.weldtool = src
-		master = F
 		add_fingerprint(user)
```

Comment:
Unrelated to your code, but I'm a bit curious why they were adding fingerprints to something they immediately qdeleted. I wonder if they meant to do `F.add_fingerprint(user)`.